From 9de492b5d7e835b086b17f8e026d2b77bfee482c Mon Sep 17 00:00:00 2001
From: Noah Lorang <noah@noahhl.com>
Date: Sun, 26 Jun 2011 09:08:33 -0400
Subject: [PATCH 2/4] Ruby 1.8.7 compatibily (round(n) doesn't exist)

---
 lib/statsd/graphite.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lib/statsd/graphite.rb b/lib/statsd/graphite.rb
index c5b34a4..136bfd5 100644
--- a/lib/statsd/graphite.rb
+++ b/lib/statsd/graphite.rb
@@ -83,7 +83,7 @@ module Statsd
       end 
       # send to graphite
       send_data stat_string
-      puts "complete. (#{time.round(3)}s)"
+      puts "complete. (#{(time*100).round()/100.0}s)"
       close_connection_after_writing
     end
   end
-- 
1.7.4.5

