diff --git a/bin/vagrant b/bin/vagrant
index fc22345..02de8f7 100755
--- a/bin/vagrant
+++ b/bin/vagrant
@@ -1,5 +1,9 @@
 #!/usr/bin/env ruby
 
+# https://bugs.gentoo.org/show_bug.cgi?id=474476
+ENV.delete('GEM_HOME')
+ENV.delete('GEM_PATH')
+
 # Trap interrupts to quit cleanly. This will be overriden at some point
 # by Vagrant. This is made to catch any interrupts while Vagrant is
 # initializing which have historically resulted in stack traces.
