diff --git a/lib_users.py b/lib_users.py
index d6e30db..2abafb3 100755
--- a/lib_users.py
+++ b/lib_users.py
@@ -26,14 +26,17 @@ def get_deleted_libs(map_file):
     """
     deletedlibs = set()
 
-    for line in map_file:
-        line = line.strip()
-        if line.endswith("(deleted)"):
-            lib = line.split()[-2]
-            is_lib = all(not fnmatch.fnmatch(lib, pattern)
-                         for pattern in NOLIBS)
-            if is_lib and lib not in deletedlibs:
-                deletedlibs.add(lib)
+    try:
+        for line in map_file:
+            line = line.strip()
+            if line.endswith("(deleted)"):
+                lib = line.split()[-2]
+                is_lib = all(not fnmatch.fnmatch(lib, pattern)
+                             for pattern in NOLIBS)
+                if is_lib and lib not in deletedlibs:
+                    deletedlibs.add(lib)
+    except IOError:
+        pass
 
     return deletedlibs
 
