diff --git a/common/macros.c b/common/macros.c
index 6cddd00..e628da8 100644
--- a/common/macros.c
+++ b/common/macros.c
@@ -2602,6 +2602,7 @@ int init_macrox_names(void){
 	add_macrox_name(MACRO_HOSTATTEMPT,"HOSTATTEMPT");
 	add_macrox_name(MACRO_NOTIFICATIONTYPE,"NOTIFICATIONTYPE");
 	add_macrox_name(MACRO_NOTIFICATIONNUMBER,"NOTIFICATIONNUMBER");
+	add_macrox_name(MACRO_NOTIFICATIONISESCALATED,"NOTIFICATIONISESCALATED");
 	add_macrox_name(MACRO_HOSTEXECUTIONTIME,"HOSTEXECUTIONTIME");
 	add_macrox_name(MACRO_SERVICEEXECUTIONTIME,"SERVICEEXECUTIONTIME");
 	add_macrox_name(MACRO_HOSTLATENCY,"HOSTLATENCY");
